File: capstone_project/ordercartservice/src/main/java/com/wipro/ordercartservice/service/OrderService.java
package com.wipro.ordercartservice.service;

import java.util.List;

import com.wipro.ordercartservice.entity.Order;

public interface OrderService {
	Order createOrder(Long userId);
    
	Order cancelOrder(Long orderId);
    
	List<Order> getAllOrders();
    
	List<Order> getOrdersByUserId(Long userId);
    
	Order getOrderById(Long orderId);

}

File: capstone_project/ecommerce/src/services/order-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Order } from '../interfaces/order';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
   private apiUrl = 'http://localhost:8083/orders'; // Corrected to 9191

  constructor(private http: HttpClient) { }

  createOrder(userId: number): Observable<Order> {
    return this.http.post<Order>(this.apiUrl, { userId });
  }

  cancelOrder(orderId: number): Observable<Order> {
    return this.http.put<Order>(`${this.apiUrl}/${orderId}/cancel`, {});
  }

  getAllOrders(): Observable<Order[]> {
    return this.http.get<Order[]>(this.apiUrl);
  }

  getOrdersByUserId(userId: number): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.apiUrl}/user/${userId}`);
  }

  getOrderById(orderId: number): Observable<Order> {
    return this.http.get<Order>(`${this.apiUrl}/${orderId}`);
  }
  
}
